packages/core/src/shared/extensions/ssh.ts

```diff
@@ -119,6 +123,26 @@ export class RemoteSshSettings extends Settings.define('remote.SSH', remoteSshTy
     }
 }
 
+export async function testSshConnection(
+    ProcessClass: typeof ChildProcess,
+    hostname: string,
+    sshPath: string,
+    user: string,
+    session: SSM.StartSessionResponse
+): Promise<void> {
+    try {
+        const env = { SESSION_ID: session.SessionId, STREAM_URL: session.StreamUrl, TOKEN: session.TokenValue }
+        await new ProcessClass(sshPath, ['-T', `${user}@${hostname}`, 'echo connected && exit']).run({
+            spawnOptions: {
+                env,
+            },
+        })
+    } catch (error) {
+        getLogger().error('SSH connection test failed: %O', error)
+        throw new SSHError('SSH connection test failed', { cause: error as Error })
```

Comment:
since we're throwing here and chaining stuff correctly, we probably don't need the log call . higher level paths , or the global error handler, should log it?

Reply:
Wanted to make no details were dropped from the logs, but with an adaption of `collectionSamErrors` (idea mentioned below) this shouldn't be a problem.